graph/graph.py

```diff
@@ -1,3 +1,69 @@
+"""
+These are classes to represent a Graph and its elements.
+It can be shared across graph algorithms.
+"""
+
+class Node(object):
+    def __init__(self, name):
+        self.name = name
+    
+    def __eq__(self, obj):
+        if isinstance(obj, Node):
+            return obj.name == self.name
+        elif isinstance(obj, str):
+            return obj == self.name
+        return False
+
+    def __repr__(self):
+        return self.name
+    
+    def __hash__(self):
+        return hash(self.name)
+
+class DirectedEdge(object):
+    def __init__(self, node_from, node_to):
+        self.nf = node_from
+        self.nt = node_to
+
+    def __eq__(self, obj):
+        if isinstance(obj, DirectedEdge):
+            return obj.nf == self.nf and obj.nt == self.nt
+        return False
+    
+    def __repr__(self):
+        return '({0} -> {1})'.format(self.nf, self.nt)
+
+class DirectedGraph(object):
+    def __init__(self, load_dict={}):
+        self.nodes = []
+        self.edges = []
+        self.adjmt = {}
+
+        if load_dict and type(load_dict) == dict:
+            for v in load_dict:
+                node_from = self.add_node(v)
+                self.adjmt[node_from] = []
+                for w in load_dict[v]:
+                    node_to = self.add_node(w)
+                    self.adjmt[node_from].append(node_to)
+                    self.add_edge(v, w)
+
+    def add_node(self, node_name):
+        try:
+            return self.nodes[self.nodes.index(node_name)]
+        except ValueError:
+            node = Node(node_name)
+            self.nodes.append(node)
+            return node
+    
+    def add_edge(self, node_name_from, node_name_to):
+        try:
+            node_from = self.nodes[self.nodes.index(node_name_from)]
+            node_to = self.nodes[self.nodes.index(node_name_to)]
+            self.edges.append(DirectedEdge(node_from, node_to))
+        except ValueError:
+            pass
+
```

Comment:
Can you comment out the lines at line number 85
```python
g = Graph(4)
g.add_edge(0, 1)
g.add_edge(0, 2)
g.add_edge(1, 2)
g.add_edge(2, 0)
g.add_edge(2, 3)
g.add_edge(3, 3)
```

Reply:
They're actually not mine but it makes sense to comment them and create test cases for this later.

Reply:
Thanks.